.evergreen/run-csfle-tests-with-mongocryptd.sh

```diff
@@ -64,12 +64,10 @@ echo "Running tests with Java ${JAVA_VERSION}"
       ${GRADLE_EXTRA_VARS} \
       --stacktrace --info --continue \
       driver-legacy:test \
-          --tests com.mongodb.ClientSideEncryptionLegacyTest \
+          --tests "*.ClientEncryption*" --tests "*.ClientSideEncryption*" \
       driver-sync:test \
-          --tests com.mongodb.client.ClientSideEncryptionTest \
-          --tests com.mongodb.client.unified.ClientSideEncryptionTest \
+          --tests "*.ClientEncryption*" --tests "*.ClientSideEncryption*" \
       driver-reactive-streams:test \
-          --tests com.mongodb.reactivestreams.client.ClientSideEncryptionTest \
-          --tests com.mongodb.reactivestreams.client.unified.ClientSideEncryptionTest \
+          --tests "*.ClientEncryption*" --tests "*.ClientSideEncryption*" \
       driver-scala:integrationTest \
-          --tests org.mongodb.scala.ClientSideEncryptionTest
+          --tests "*.ClientEncryption*" --tests "*.ClientSideEncryption*"
```

Comment:
This change together with the `assumeFalse/assumeTrue(CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE != null)` checks in `AbstractClientSideEncryptionNotSpawnMongocryptdTest` solves the issue discussed in https://mongodb.slack.com/archives/CFPHLTRMK/p1683160528003019: some of the tests added in this PR not only do not require `crypt_shared`, but require the library not to be loaded, which is why they are run separately (in different process) from those that load `crypt_shared` (once it is loaded, it can't be unloaded).